src/app/backend/handler/apihandler.go

```diff
@@ -527,13 +526,39 @@ func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.Hea
 	return wsContainer
 }
 
+func getApiCient(clientconfig clientcmd.ClientConfig, request *restful.Request) (*clientK8s.Clientset, error) {
+	cfg, err := clientconfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	token := request.HeaderParameter("Authorization")
+	if strings.HasPrefix(token, "Bearer ") {
+		cfg.BearerToken = strings.TrimPrefix(token, "Bearer ")
+	}
+
+	apiclient, err := clientK8s.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return apiclient, nil
+}
+
 // Handles get pet set list API call.
 func (apiHandler *APIHandler) handleGetStatefulSetList(request *restful.Request,
 	response *restful.Response) {
 	namespace := parseNamespacePathParameter(request)
 	dataSelect := parseDataSelectPathParameter(request)
 	dataSelect.MetricQuery = dataselect.StandardMetrics
-	result, err := statefulsetlist.GetStatefulSetList(apiHandler.client, namespace, dataSelect, &apiHandler.heapsterClient)
+
+	apiclient, err := getApiCient(apiHandler.clientConfig, request)
+	if err != nil {
+		handleInternalError(response, err)
+		return
+	}
```

Comment:
hmm, shame that this requires so much code duplication. Is it possible to abstract this away in a wrapping function?